dvc/dependency/repo.py

```diff
@@ -114,7 +114,9 @@ def _make_fs(
             remote_config = None
 
         conf = self.def_repo.get("config")
-        if conf:
+        if isinstance(conf, dict):
+            config = conf  # type: ignore[unreachable]
```

Comment:
Also need to update the type for `def_repo` from `Dict[str, str]` instead of ignoring here.

Reply:
You can either use a `TypedDict` or just type it as `Dict[str, Any]` (and lose type of values).